### PR TITLE
Grab refactor and species hooks

### PR DESCRIPTION
**Pull request by BothNevarBackwards**
Grabs should work pretty much the same as they do now, just with a more organised code side. Ported from baystation.

### CODE REVIEW

File: code/game/gamemodes/technomancer/spells/apportation.dm [hunk 1]
```diff
@@ -57,22 +57,17 @@
 			L.throw_at(get_step(get_turf(src),get_turf(L)), 4, 1, src)
 			user.drop_item(src)
 			src.loc = null
+			if(istype(user, /mob/living))
+				var/mob/living/U = user
+				spawn(1 SECOND)
+					if(!user.Adjacent(L))
+						user << "<span class='warning'>\The [L] is out of your reach.</span>"
```
Comment (Belsima) [inline code/game/gamemodes/technomancer/spells/apportation.dm]: to_chat

**Comment by Belsima** [comment]
i love you thank you for doing my work for me

**Comment by BothNevarBackwards** [comment]
Which work is that?

**Comment by BothNevarBackwards** [comment]
@Belsima fixed

**Comment by Belsima** [comment]
i was gonna get to this eventually but you made life easy instead

**Comment by Anewbe** [comment]
Sorry for the delay on this, what precisely is changed by this PR? Can you give me a specific list?

**Comment by chaoko99** [comment]
I feel the need to put it out there that tabling and wallslamming et al do not work via this system without reimplementation. It's weird.

**Comment by Anewbe** [comment]
Author hasn't been active since the PR was made, and it conflicts. Closing.

**Pull request closed without merging**